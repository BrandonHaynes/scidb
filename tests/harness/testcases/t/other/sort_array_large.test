# sort a large enough array that multiple sorted runs are created

--setup
--start-query-logging
--start-igdata

create array A <d:int64> [ X=0:40000000,1000000,0]
store(build(A, random()), A)

--test

store(sort(A, d), ASort)
store(subarray(ASort, 1000, 1100), ASortSlice1)
store(subarray(ASort, 100000, 100100), ASortSlice2)
store(subarray(ASort, 500000, 500100), ASortSlice3)
store(window(ASortSlice1, 0, 1, min(d)), ASortWindow1)
store(window(ASortSlice2, 0, 1, min(d)), ASortWindow2)
store(window(ASortSlice3, 0, 1, min(d)), ASortWindow3)

--stop-igdata

aggregate(filter(join(ASortSlice1, ASortWindow1), d!=d_min), count(*))
aggregate(filter(join(ASortSlice2, ASortWindow2), d!=d_min), count(*))
aggregate(filter(join(ASortSlice3, ASortWindow3), d!=d_min), count(*))

--cleanup

remove(A)
remove(ASort)
remove(ASortSlice1)
remove(ASortSlice2)
remove(ASortSlice3)
remove(ASortWindow1)
remove(ASortWindow2)
remove(ASortWindow3)

--stop-query-logging




