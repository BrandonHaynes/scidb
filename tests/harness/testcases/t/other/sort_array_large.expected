SCIDB QUERY : <create array A <d:int64> [ X=0:40000000,1000000,0]>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(build(A, random()), A)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(sort(A, d), ASort)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(subarray(ASort, 1000, 1100), ASortSlice1)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(subarray(ASort, 100000, 100100), ASortSlice2)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(subarray(ASort, 500000, 500100), ASortSlice3)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(window(ASortSlice1, 0, 1, min(d)), ASortWindow1)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(window(ASortSlice2, 0, 1, min(d)), ASortWindow2)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(window(ASortSlice3, 0, 1, min(d)), ASortWindow3)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <aggregate(filter(join(ASortSlice1, ASortWindow1), d!=d_min), count(*))>
{i} count
{0} 0

SCIDB QUERY : <aggregate(filter(join(ASortSlice2, ASortWindow2), d!=d_min), count(*))>
{i} count
{0} 0

SCIDB QUERY : <aggregate(filter(join(ASortSlice3, ASortWindow3), d!=d_min), count(*))>
{i} count
{0} 0

SCIDB QUERY : <remove(A)>
Query was executed successfully

SCIDB QUERY : <remove(ASort)>
Query was executed successfully

SCIDB QUERY : <remove(ASortSlice1)>
Query was executed successfully

SCIDB QUERY : <remove(ASortSlice2)>
Query was executed successfully

SCIDB QUERY : <remove(ASortSlice3)>
Query was executed successfully

SCIDB QUERY : <remove(ASortWindow1)>
Query was executed successfully

SCIDB QUERY : <remove(ASortWindow2)>
Query was executed successfully

SCIDB QUERY : <remove(ASortWindow3)>
Query was executed successfully

